app/scripts/modules/amazon/src/serverGroup/configure/wizard/pages/ServerGroupInstanceType.tsx

```diff
@@ -26,34 +42,56 @@ export class ServerGroupInstanceType
     // Instance profile is already set on values.viewState, so just use that value.
     // Once angular is gone from this component tree, we can move all the viewState stuff
     // into react state
-    this.setState({});
+    this.setState({ newProfileType: _profile, newInstanceType: undefined });
   };
 
   private instanceTypeChanged = (type: string) => {
     const { values } = this.props.formik;
     values.instanceTypeChanged(values);
     this.props.formik.setFieldValue('instanceType', type);
+    this.setState({ newInstanceType: type, newProfileType: undefined });
+  };
+
+  private setUnlimitedCpuCredits = (unlimitedCpuCredits: boolean | undefined) => {
+    if (this.props.formik.values.unlimitedCpuCredits !== unlimitedCpuCredits) {
+      this.props.formik.setFieldValue('unlimitedCpuCredits', unlimitedCpuCredits);
+      this.setState({});
+    }
   };
 
   public render() {
     const { values } = this.props.formik;
-    const showTypeSelector = !!(values.viewState.disableImageSelection || values.amiName);
 
     const { InstanceArchetypeSelector, InstanceTypeSelector } = NgReact;
+    const showTypeSelector = !!(values.viewState.disableImageSelection || values.amiName);
+
+    const isLaunchTemplatesEnabled = AWSProviderSettings.serverGroups?.enableLaunchTemplates;
```

Comment:
A different setting should be used here. There should be a specific flag to enable CPU credits. By this logic, my users would see this toggle even though we do not want them to.

`AWSProviderSettings.serverGroups?.enableCpuCredits`

Reply:
I didn't add a separate setting thinking this feature should be tied to `enableLaunchTemplates`. 

However, it makes sense to create the separate setting. Could you confirm if the following behaviors match your expectations. 
- When the setting is turned off, by default, t2/t3 instances will be launched with `standard` cpu credits (due to the defaults in clouddriver). The server group details will show cpu credits as `standard` in `launch template` details even when `enableCpuCredits` is turned off.
- Users will have to explicitly turn on the `enableCpuCredits` setting to choose `unlimited`.